src/Language/PureScript/Types.hs

```diff
@@ -162,14 +168,14 @@ replaceAllTypeVars = go []
   genName orig inUse = try' 0
     where
     try' :: Integer -> Text
-    try' n | (orig <> show n) `elem` inUse = try' (n + 1)
-          | otherwise = orig <> show n
+    try' n | (orig <> T.pack (show n)) `elem` inUse = try' (n + 1)
+           | otherwise = orig <> T.pack (show n)
 
 -- |
 -- Collect all type variables appearing in a type
 --
 usedTypeVariables :: Type -> [Text]
-usedTypeVariables = ordNub . everythingOnTypes (++) go
+usedTypeVariables = nub . everythingOnTypes (++) go
```

Comment:
I just reversed the changes I made when introducing a project specific prelude based on Protolude. That's for a different PR and needs some discussion first.